### PR TITLE
Improve error handling

### LINKED ISSUE #36: bulk index method invokes on data for status 400
Possibly a bug.

Expected behavior - on error gets invoked

### PR DESCRIPTION
**Pull request by dhruvdutt**
Fixes #36 [#163](https://github.com/appbaseio/reactivesearch/issues/163)

### CODE REVIEW

**Comment by divyanshu013** [comment]
Why is the `fetch` request not getting caught in case of status >= 400?

**Comment by dhruvdutt** [comment]
@divyanshu013 I think that's how it works. An error is not an exception. :smile: 

Further reading: [#18](https://github.com/whatwg/fetch/issues/18), [#203](https://github.com/github/fetch/issues/203).

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
 	"name": "appbase-js",
-	"version": "2.3.0",
+	"version": "2.3.1",
```
Comment (dhruvdutt) [inline package.json]: Would this be a patch release or something else?

I hope it doesn't break things for those who depends on status 400 inside `.on('data')`.

cc @divyanshu013 :yum:

Comment (divyanshu013) [reply package.json]: Please do a major release 🙏

**Pull request merged**